### PR TITLE
Feature/fga/remove from main thread

### PR DESCRIPTION
**Pull request by ganfra**
Remove some code from main thread

### CODE REVIEW

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt [hunk 1]
```diff
@@ -122,17 +122,12 @@ class MessagesPresenter @AssistedInject constructor(
...
         val inviteProgress = remember { mutableStateOf<Async<Unit>>(Async.Uninitialized) }
-
-        val showReinvitePrompt by remember(
-            hasDismissedInviteDialog,
-            composerState.hasFocus,
-            syncUpdateFlow,
-        ) {
-            derivedStateOf {
-                !hasDismissedInviteDialog && composerState.hasFocus && room.isDirect && room.activeMemberCount == 1L
+        var showReinvitePrompt by remember { mutableStateOf(false) }
```
Comment (jmartinesp) [inline features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt:125]: Is this change needed? The check seems really fast to me, and using the `LaunchedEffect` will trigger a recomposition, right? The `room.activeMemberCount` does come from Rust, but the originating code is just a sum of 2 numbers and it's sync AFAIK, so it should be fast.

Comment (ganfra) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt:125]: It's hitting the db underneath if I'm not mistaken

Comment (jmartinesp) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt:125]: Oh, you're right, it's even async under the hood. I thought we mapped that instead of calling the underlying function each time. This change makes sense then, thanks!

**Review by jmartinesp — APPROVED** [review]
LGTM! There is a warning about unused imports.

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt (hunk 1 above)
Comment (ganfra) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt:125]: Ok, I'm mistaken, let's revert this one then.

Comment (ganfra) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt:125]: Actually, I think almost all methods are async in the rust layer, but exposed as sync...
Using `val` in our layer is then misleading even more, we should probably use `suspend fun` at some point.
